apps/greencheck/tests/test_azure_cloud_importer.py

```diff
@@ -0,0 +1,114 @@
+import pytest
+import pathlib
+import ipaddress
+import json
+from io import StringIO
+
+from django.core.management import call_command
+from apps.greencheck.management.commands import update_azure_ip_ranges
+from apps.accounts.models import Hostingprovider
+from apps.greencheck.models import GreencheckIp
+from apps.greencheck.management.commands.update_azure_ip_ranges import GREEN_REGIONS
+
+from django.conf import settings
+
+
+@pytest.fixture
+def hosting_provider():
+
+    # oregon, *rest = [region for region in GREEN_REGIONS if region[1] == "us-west-2"]
+    name, region, host_id = ("Azure US West", "az-west-2", 123)
+    return Hostingprovider(
+        archived=False,
+        country="US",
+        customer=False,
+        icon="",
+        iconurl="",
+        id=host_id,
+        model="groeneenergie",
+        name=name,
+        partner="",
+        showonwebsite=True,
+        website="http://azure.microsoft.com",
+    )
+
+
+@pytest.fixture
+def azure_cloud_provider(hosting_provider):
+    hosting_provider.save()
+    return update_azure_ip_ranges.MicrosoftCloudProvider()
+
+
+@pytest.fixture
+def azure_json_ip_ranges():
+    this_file = pathlib.Path(__file__)
+    json_path = this_file.parent.parent.joinpath("fixtures", "azure_ip_ranges.json")
+    with open(json_path) as ipr:
+        ip_ranges = json.loads(ipr.read())
+        return ip_ranges
+
+
+@pytest.mark.django_db
+class TestAZURECLoudImporter:
+    def test_fetch_ip(self, hosting_provider, azure_cloud_provider):
+        """
+        Do we fetch the data from Azure?
+        """
+        hosting_provider.save()
+        res = azure_cloud_provider.retrieve()
+
+        assert len(res.keys()) == 3 # TODO: Figure out what value this should be
+
+    def pullout_green_regions(self, hosting_provider, azure_cloud_provider):
+        res = azure_cloud_provider.retrieve()
+
+        _, region, host_id = azure_cloud_provider.green_regions[0]
+
+        iprs = azure_cloud_provider.pullout_green_regions(res, region)
+
+        ip_ranges = azure_cloud_provider.ip_ranges_for_hoster(iprs)
+
+        assert isinstance(ip_ranges[0], ipaddress.IPv4Network)
+
+    def test_update_hoster(self, hosting_provider, azure_cloud_provider):
+
+        res = azure_cloud_provider.retrieve()
+        _, region, host_id = ("Azure US West", "az-west-2", settings.AZURE_PROVIDER_ID)
+        ip_ranges = azure_cloud_provider.convert_to_networks(res)
+
+        ip_start, ip_end = ip_ranges[0][0], ip_ranges[0][-1]
+        assert GreencheckIp.objects.all().count() == 0
+
+        hosting_provider.save()
+
+        # create one new green IP range
+        azure_cloud_provider.update_hoster(hosting_provider, ip_start, ip_end)
+
+        assert GreencheckIp.objects.all().count() == 1
+
+    def test_process(
```

Comment:
would you please add a longer method name here, and/or add a docstring about what process we're testing? It wasn't obvious to me, and it's one of the first places we'd look when trying to understand what it would mean if this test was failing